internal/storage/conformance/conformance.go

```diff
@@ -0,0 +1,611 @@
+				PeerName:  storage.Wildcard,
+				Namespace: storage.Wildcard,
+			},
+			namePrefix: "a",
+			results: []*pbresource.Resource{
+				seedData[0],
+				seedData[1],
+				seedData[3],
+				seedData[5],
+				seedData[6],
+			},
+		},
+	}
+
+	t.Run("List", func(t *testing.T) {
+		backend := opts.NewBackend(t)
+		ctx := testContext(t)
+
+		for _, r := range seedData {
+			_, err := backend.WriteCAS(ctx, r, "")
+			require.NoError(t, err)
+		}
+
+		for desc, tc := range testCases {
+			t.Run(desc, func(t *testing.T) {
+				res, err := backend.List(ctx, tc.resourceType, tc.tenancy, tc.namePrefix)
+				require.NoError(t, err)
+				require.ElementsMatch(t, res, tc.results)
+			})
+		}
+	})
+
+	t.Run("WatchList", func(t *testing.T) {
+		for desc, tc := range testCases {
+			t.Run(fmt.Sprintf("%s - initial snapshot", desc), func(t *testing.T) {
+				backend := opts.NewBackend(t)
+				ctx := testContext(t)
+
+				// Write the seed data before the watch has been established.
+				for _, r := range seedData {
+					_, err := backend.WriteCAS(ctx, r, "")
+					require.NoError(t, err)
+				}
+
+				watch, err := backend.WatchList(ctx, tc.resourceType, tc.tenancy, tc.namePrefix)
+				require.NoError(t, err)
+
+				for i := 0; i < len(tc.results); i++ {
+					ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+					t.Cleanup(cancel)
+
+					event, err := watch.Next(ctx)
+					require.NoError(t, err)
+
+					require.Equal(t, pbresource.WatchEvent_OPERATION_UPSERT, event.Operation)
+					require.Containsf(t, tc.results, event.Resource, "resource not in expected results: %s", event.Resource.Id)
+				}
+			})
+
+			t.Run(fmt.Sprintf("%s - following events", desc), func(t *testing.T) {
+				backend := opts.NewBackend(t)
+				ctx := testContext(t)
+
+				watch, err := backend.WatchList(ctx, tc.resourceType, tc.tenancy, tc.namePrefix)
+				require.NoError(t, err)
+
+				// Write the seed data after the watch has been established.
+				for _, r := range seedData {
+					_, err := backend.WriteCAS(ctx, r, "")
+					require.NoError(t, err)
+				}
+
+				for i := 0; i < len(tc.results); i++ {
+					ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+					t.Cleanup(cancel)
+
+					event, err := watch.Next(ctx)
+					require.NoError(t, err)
+
+					require.Equal(t, pbresource.WatchEvent_OPERATION_UPSERT, event.Operation)
+					require.Containsf(t, tc.results, event.Resource, "resource not in expected results: %s", event.Resource.Id)
+				}
+
+				// Delete a random resource to check we get an event.
+				del := tc.results[rand.Intn(len(tc.results))]
+				require.NoError(t, backend.DeleteCAS(ctx, del.Id, del.Version))
+
+				ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+				t.Cleanup(cancel)
+
+				event, err := watch.Next(ctx)
+				require.NoError(t, err)
+
+				require.Equal(t, pbresource.WatchEvent_OPERATION_DELETE, event.Operation)
+				require.Equal(t, del, event.Resource)
+			})
+		}
+	})
+}
+
+func testWatch(t *testing.T, opts TestOptions) {
```

Comment:
I'm not seeing where this function is getting called.

Reply:
Whoops! Good catch. Got left behind in a refactoring 🤦🏻